### ISSUE TITLE
Configuration used to start a profile is not saved

### ISSUE THREAD

**Issue by jorgemoralespou** [minishift/minishift#2611]
We need to persist the configuration used for a profile (at least the relative to driver, memory, cpu, etc....) as in a second start a user might not know all the parameters he used to create the profile and he can mess it up.

Background:

I provisioned a fedora VM on Virtualbox, to use the generic driver.
I provisioned the cluster with:

```
minishift start --profile remote-10-10-10-10 --vm-driver generic --remote-ipaddress 10.10.10.10 --public-hostname 10.10.10.10 --remote-ssh-user vagrant --remote-ssh-key ./.vagrant/machines/default/virtualbox/private_key
```

I used the cluster for a while. Note that I used a profile named "remote-10-10-10-10" so that if I start that profile I expect the same configuration to be used.

I stop the cluster, and then start it again:

```
$ minishift start --profile remote-10-10-10-10
-- Starting profile 'remote-10-10-10-10'
-- Check if deprecated options are used ... OK
-- Checking if https://github.com is reachable ... OK
-- Checking if requested OpenShift version 'v3.9.0' is valid ... OK
-- Checking if requested OpenShift version 'v3.9.0' is supported ... OK
-- Checking if requested hypervisor 'xhyve' is supported on this platform ... OK
-- Checking if xhyve driver is installed ...
   Driver is available at /usr/local/bin/docker-machine-driver-xhyve
   Checking for setuid bit ... OK
-- Checking the ISO URL ... OK
-- Checking if provided oc flags are supported ... OK
-- Starting the OpenShift cluster using 'xhyve' hypervisor ...
-- Starting Minishift VM ..................................................... FAIL E0717 18:45:05.309406   51049 start.go:416] Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded. Retrying.
Error starting the VM: Error configuring authorization on host: Maximum number of retries (60) exceeded
```

I think I've said this before, but configuration needs to be saved in the profile, so that I can just restart the profile with a minimal command, allowing for overriding those values that could be overridden, but if this is difficult, better to not override anything than not save start configuration.

**Comment by praveenkumar** [comment]
@jorgemoralespou one way to resolve this is to put those remote options in the config file itself like we do for cpus, memory ..etc and don't use any of the options during the start.

```
$ minishift profile set remote-10-10-10-10
$ minishift config set remote-ipaddress 10.10.10.10
$ minishift config set remote-ssh-user vagrant
$ minishift config set remote-ssh-key ./.vagrant/machines/default/virtualbox/private_key
$ minishift config set vm-driver genric
```

**Comment by jorgemoralespou** [comment]
@praveenkumar I know the workarounds, but IMHO this is not the correct experience. That configuration should be preserved by default.
Are you saying that this is not an issue because it can be saved in the configuration using a miriad of config set commands?

**Comment by praveenkumar** [comment]
@jorgemoralespou Till now we always say to use same options for `start` command if you do start => stop => start this is not any different and most of the time a user uses `config` to persist this.

**Comment by jorgemoralespou** [comment]
@praveenkumar and I've always said it's a bad experience 😄
When you use multiple flags in start, you have to remember them all. If one needs to use the config file to avoid remembering these flags, why don't saving them in config on first start?
I think that your answer confirms that this should be the default behavior.

**Comment by jorgemoralespou** [comment]
If you all guys oppose to fixing this, at least provide a --persist flag to start that will do it when passed. Not ideal, but better.

**Comment by praveenkumar** [comment]
@jorgemoralespou sure, we will try to handle it soon, one way or other but I think there is already existing issue, will check and put it here.

**Comment by LalatenduMohanty** [comment]
> When you use multiple flags in start, you have to remember them all. If one needs to use the config file to avoid remembering these flags, why don't saving them in config on first start?

This might be a better idea than `--persist` flag. At least make things consistent e.g. `minishift config view` will return what flags users had run minishift with.  Basically we  will save the viper config in config.json during every start.  Need to check if it will create any complications.

**Comment by gbraad** [comment]
Related to an old issue #179. As mentioned on that issue, it is best to print out the ignored options and what is run, and/or save the sane expectation. Options we provide on start aren't supposed to be a one-off use.

**Comment by LalatenduMohanty** [comment]
Resolved via #2634
